Xwt.Gtk/Xwt.GtkBackend/TableViewBackend.cs

```diff
@@ -514,6 +519,40 @@ static CustomTreeView ()
 				GtkWorkarounds.RemoveKeyBindingFromClass (Gtk.TreeView.GType, Gdk.Key.BackSpace, Gdk.ModifierType.None);
 		}
 
+		protected override bool OnButtonPressEvent (EventButton evnt)
+		{
+			// If we are clicking on already selected row, delay the selection until we are certain that
+			// the user is not starting a DragDrop operation. 
+			// This is needed to allow user to drag multiple selected rows.
+			TreePath treePath;
+			TreeViewColumn column;
+			GetPathAtPos ((int)evnt.X, (int)evnt.Y, out treePath, out column);
+			var ctrlShiftMask = (evnt.State & (Gdk.ModifierType.ShiftMask | Gdk.ModifierType.ControlMask | Gdk.ModifierType.Mod2Mask));
+			if (evnt.Button == 1 && this.Selection.PathIsSelected (treePath) && ctrlShiftMask == 0) {
+				delayedSelection = treePath;
+				delayedSelectionColumn = column;
+				delayedSelectionX = evnt.X;
+				delayedSelectionY = evnt.Y;
+				Selection.SelectFunction = (_, __, ___, ____) => false;
+				var result = base.OnButtonPressEvent (evnt);
+				Selection.SelectFunction = (_, __, ___, ____) => true;
```

Comment:
I think we should probably restore the previous value for `Selection.SelectFunction` here, in case it had been set to something custom. Also, this should be done in a finally block in case the base handler happens to throw an exception.

Reply:
We can't save previous `SelectFunction` value here because this property has no getter 😫

Reply:
Ah I see. Well it's probably ok because xwt owns the `CustomTreeView` anyway.